### PR TITLE
refactor(kpi-controller): fix type errors and lint issues

### PR DESCRIPTION
**Pull request by Samyak2**
- added `Optional[]` to types where missing
- added missing docstrings
- fixed type issues in `run_anomaly_for_kpi`
    - renamed `end_date` to help type checker recognize correctly
    - added an exception on `else:`
- added ignores to SQLalchemy filters where applicable
- added missing type annotations to `get_anomaly_count`
- formatted using `isort` and `black`

### CODE REVIEW

File: chaos_genius/controllers/kpi_controller.py [hunk 1]
```diff
@@ -58,45 +53,47 @@ def get_kpi_data_from_id(n: int) -> dict:
...
-        and kpi_info["scheduler_params"]["scheduler_frequency"] == "H"
+        end_date is None and kpi_info["scheduler_params"]["scheduler_frequency"] == "H"
     ):
-        end_date = datetime.today().date()
+        true_end_date = datetime.today().date()
+
+    else:
+        raise Exception(
+            "end_date isn't provided and scheduler_frequency is not recognized."
+        )
```
Comment (kartikay-bagla) [inline chaos_genius/controllers/kpi_controller.py:84]: I don't think we've handled the case where end_date is not None here. Here's my solution:
```python
if end_date is None:
    if scheduler_freq = "D":
        # create true_end_date for daily
    elif scheduler_freq = "H":
        # create true_end_date for hourly
    else:
        # raise error
else:
    true_end_date = end_date

Comment (kartikay-bagla) [inline chaos_genius/controllers/kpi_controller.py:84]: I don't think we've handled the case where end_date is not None here. Here's my solution:
```python
if end_date is None:
    if scheduler_freq = "D":
        # create true_end_date for daily
    elif scheduler_freq = "H":
        # create true_end_date for hourly
    else:
        # raise error
else:
    true_end_date = end_date
```

**Review by kartikay-bagla — CHANGES_REQUESTED** [review]

File: chaos_genius/controllers/kpi_controller.py (hunk 1 above)
Comment (Samyak2) [reply chaos_genius/controllers/kpi_controller.py]: Good catch. Fixed in 261fd0fdca7353234959e11734bd14005744f923

**Review by kartikay-bagla — APPROVED** [review]
LGTM

**Pull request merged**